lib/ar_lazy_preload/active_record/relation.rb

```diff
@@ -1,12 +1,26 @@
 # frozen_string_literal: true
 
 require "ar_lazy_preload/context"
+require "ar_lazy_preload/contexts/temporary_preload_config"
 
 module ArLazyPreload
   # ActiveRecord::Relation patch with lazy preloading support
   module Relation
     attr_writer :preloads_associations_lazily
 
+    def preload_associations(records)
```

Comment:
IMO it looks a bit clearer (since we instantiate preloader once):

```ruby
preload = preload_values
preload += includes_values unless eager_loading?
return if preload.empty?

preloader = build_preloader
preload.each do |associations|
  preloader_associations = preloader.preload records, associations
  preloader_associations.each do |preloader_association|
    handle_preloaded_records(preloader_association.preloaded_records)
  end
end
```

Reply:
I don't have a strong opinion here, my idea was that It was [copy-paste from the rails code](https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/relation.rb#L742-L750) and I think it's easier to track the difference if needed. But I'm fine to change to your option.

Reply:
Sounds reasonable, let's keep it